hafnian/_hermite_multidimensional.py

```diff
@@ -0,0 +1,150 @@
+from ._hafnian import input_validation
+
+
+def density_matrix(A, d, resolution, renorm=False):
+    r"""Returns photon number statistics of a Gaussian state for a given covariance matrix `A`.
+
+
+    Args:
+        A (array): a square, symmetric :math:`N\times N` array.
+        d (array): a one diemensional size :math:`N` array.
+        resolution (int): highest number of photons to be resolved.
+
+    """
+
+    dim = A.shape[0]
+    U = np.eye(dim, dtype=complex)
+
+    for i in range(dim // 2):
+        U[i, i] = 0 - 1j
+        U[i + dim // 2, i] = 1 + 0 * 1j
+        U[i, i + dim // 2] = 0 + 1j
+
+    U = U / np.sqrt(2)
+
+    U3 = U.transpose()
+    U1 = U3.conjugate()
+    U2 = U.conjugate()
+
+    tmp1 = np.eye(dim, dtype=complex) + 2 * A
+    tmp1_inv = np.linalg.inv(tmp1)
+
+    tmp2 = np.eye(dim, dtype=complex) - 2 * A
+    tmp2_inv = np.linalg.inv(tmp2)
+
+    tmp = tmp1_inv @ U2
+    tmp = tmp2 @ tmp
+    R = U1 @ tmp
+
+    tmpy = tmp2_inv @ d
+    y = U3 @ tmpy
+    y = 2 * y
+
+    return hm(R, y, resolution, ren=renorm)
+
+
+def hermite_multidimensional(R, resolution, y=None, renorm=False, make_tensor=True):
+    r"""
+    Returns the multidimensional Hermite polynomials :math:`H_k^{(R)}(y)` where :math:`R` is an :math:n \times n: square matrix,
+    :math:`y` is an :math:`n` dimensional vector. The polynomials are parametrized by the multi-index :math:`k=(k_0,k_1,\ldots,k_{n-1})
+    and are calculated for all values :math:`0 \leq k_j < \text{resolution}`.
+    Thus a tensor of dimensions :math:`\text{resolution}^n` is returned. This tensor can either be flattened into a vector or returned as an
+    actual tensor with :math:`n` indices.
+    Note that is R = np.array([[1]]) then :math:`H_k^{(R)}(y)` are precisely the well known **probabilists' Hermite polynomials** :math:`He_k(y)`:,
+    and if R = 2*np.array([[1]]) then :math:`H_k^{(R)}(y)` are precisely the well known **physicists' Hermite polynomials** :math:`H_k(y)`:.
+    Args:
+        R (array): Square matrix parametrizing the Hermite polynomial family
+        resolution (int): Maximum size of the subindices in the Hermite polynomial
+        y (array): Vector for the argument of the Hermite polynomial
+        renorm (bool): If True returns :math:`H_k^{(R)}(y)/\prod(\prod_i k_i!)`
+        make_tensor: If False returns a flattened one dimensional array instead of a tensor with the values of the polynomial
+    Returns:
+        (array): The multidimensional Hermite polynomials.
+    """
+    input_validation(R, check_symmetry=False)
+    n, _ = R.shape
+    if y is None:
+        y = np.zeros([n], dtype=complex)
+
+    m = y.shape[0]
+    if m != n:
+        raise ValueError("The matrix R and vector y have incompatible dimensions")
+
+    values = np.array(hm(R, y, resolution, ren=renorm))
+
+    if make_tensor:
+        shape = resolution * np.ones([n], dtype=int)
+        values = np.reshape(values, shape)
+
+    return values
```

Comment:
Does this function do what https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.polynomial.hermite.hermval.html does?

Reply:
@nquesada ?